simtools/corsika/corsika_output.py

```diff
@@ -0,0 +1,1301 @@
+        """
+        self.individual_telescopes = individual_telescopes
+        self.num_of_hist = len(self.telescope_indices) if self.individual_telescopes is True else 1
+
+        self.hist_position, self.hist_direction, self.hist_time_altitude = [], [], []
+
+        for _ in range(self.num_of_hist):
+            boost_axes_position = self._create_regular_axes("hist_position")
+            self.hist_position.append(
+                bh.Histogram(
+                    boost_axes_position[0],
+                    boost_axes_position[1],
+                    boost_axes_position[2],
+                )
+            )
+            boost_axes_direction = self._create_regular_axes("hist_direction")
+            self.hist_direction.append(
+                bh.Histogram(
+                    boost_axes_direction[0],
+                    boost_axes_direction[1],
+                )
+            )
+
+            boost_axes_time_altitude = self._create_regular_axes("hist_time_altitude")
+            self.hist_time_altitude.append(
+                bh.Histogram(
+                    boost_axes_time_altitude[0],
+                    boost_axes_time_altitude[1],
+                )
+            )
+
+    def _fill_histograms(self, photons, azimuth_angle=None, zenith_angle=None):
+        """Fill all the histograms created by self._create_histogram with the information of the
+         photons on the ground.
+         If the azimuth and zenith angles are provided, the Cherenkov photon's coordinates are
+         filled in the plane perpendicular to the incoming direction of the particle.
+
+        Parameters
+        ----------
+        photons: list
+            List of size M of numpy.array of size (N,8), where M is the number of telescopes in the
+            array, N is the number of photons that reached each telescope. The following information
+             of the Cherenkov photons on the ground are saved:
+             x: x position on the ground (CORSIKA coordinate system),
+             y: y position on the ground (CORSIKA coordinate system),
+             cx: direction cosinus in the x direction, i.e., the cosinus of the angle between the
+             incoming direction and the x axis,
+             cy: direction cosinus in the y direction, i.e., the cosinus of the angle between the
+             incoming direction and the y axis,
+             time: time of arrival of the photon in ns. The clock starts when the particle crosses
+             the top of the atmosphere (CORSIKA-defined) if `self.event_first_interaction_heights`
+             is positive or at first interaction if otherwise.
+             zem: altitude where the photon was generated in cm,
+             photons: number of photons associated to this bunch,
+             wavelength: the wavelength of the photons in nm.
+        azimuth_angle: astropy.Quantity
+            Azimuth angle to rotate the observational plane and obtain it perpendicular to the
```

Comment:
Don't get this sentence. Is the grammar correct for `..and obtain it perpendicular...`?

Reply:
I improved it now and changed the name of the variables to match the ones from the rotate function.
I am actually not sure what would be best, to rotate the observational plane by default and align it to the incoming direction of particles or not to rotate by default.